crowbar_framework/lib/repository.rb

```diff
@@ -0,0 +1,144 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Repository
+  class Check
+    attr_reader :platform, :id, :name, :required, :available, :md5, :url
+
+    def initialize(platform, repo)
+      @platform = platform
+      @id = repo
+      @name = config_repo["name"]
+      @required = config_requirement
+      @available = validate_repo_dir
+      @md5 = validate_key_file
+      @url = config_web_url
+    end
+
+    protected
+
+    #
+    # config
+    #
+    def config
```

Comment:
This is an example of why I think the current class is the wrong thing.  You have an instance of a `Repository::Check` object, and it's loading the YAML for *all* repositories.  That means you'll parse the same YAML file every single time you do a check.  If there are 10 repos and each repo requires 2 checks (e.g. the md5 and the key) then you will parse the same YAML file 20 times!

Instead I recommend a memoizing class method `all` which loads the repository registry into a class variable.  Then the constructor can simply look up the right repository from that.